swift/ql/test/library-tests/dataflow/taint/data.swift

```diff
@@ -0,0 +1,25 @@
+
+class Data
+{
+    init<S>(_ elements: S) {}
+}
+
+func source() -> String { return "" }
+func sink(arg: Data) {}
+func sink2(arg: String) {}
+
+func taintThroughData() {
+	let dataClean = Data("123456".utf8)
+	let dataTainted = Data(source().utf8)
+	let dataTainted2 = Data(dataTainted)
+
+	sink(arg: dataClean)
+	sink(arg: dataTainted) // tainted [NOT DETECTED]
```

Comment:
Now that we have InlineExpectationsTest for Swift we should convert out tests to use that framework. No need to do it in this PR, though.

Reply:
Yeah, that might be good as a follow-up PR.

Reply:
I can make an inline expectation test for the sinks that taint reaches (annotated with where taint came from), but it won't have the same detail (path edges etc) as the query as it is.  Do you think I should keep both queries or prefer just one?

Reply:
Yeah, I think there's value in testing the paths as well.

Reply:
I'm a little reluctant to have three similar but different tests on the same code (when you include `LocalTaint.ql`), but the taint tests are really important.  :+1:

Reply:
https://github.com/github/codeql/pull/10014